simplified-app-shared/src/main/java/org/nypl/simplified/app/AnnotationsManager.kt

```diff
@@ -0,0 +1,600 @@
+package org.nypl.simplified.app
+
+import android.app.AlertDialog
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import com.android.volley.*
+import com.android.volley.Response.Listener
+import com.android.volley.toolbox.Volley
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import com.io7m.jfunctional.Some
+import org.json.JSONException
+import org.json.JSONObject
+import org.joda.time.Instant
+import org.nypl.drm.core.AdobeDeviceID
+import org.nypl.simplified.app.catalog.annotation.AnnotationResult
+import org.nypl.simplified.app.reader.ReaderBookLocation
+import org.nypl.simplified.books.core.AccountCredentials
+import org.nypl.simplified.books.core.AccountsControllerType
+import org.nypl.simplified.multilibrary.Account
+import org.nypl.simplified.volley.NYPLJsonObjectRequest
+import org.slf4j.LoggerFactory
+import java.io.IOException
+
+/**
+ * Performs work relevant to syncing OPDS Annotations.
+ * Current features for books:
+ * 1 - Current reading position
+ * 2 - Bookmarks saved by the user
+ */
+class AnnotationsManager(private val libraryAccount: Account,
+                         private val credentials: AccountCredentials,
+                         private val context: Context)
+{
+  private val requestQueue = Volley.newRequestQueue(context)
+
+  private companion object {
+    val LOG = LoggerFactory.getLogger(AnnotationsManager::class.java)!!
+  }
+
+  /**
+   * Queries the server, and based on a set of conditions, decides whether
+   * or not to inform the user about "Syncing" in an Alert Dialog, where
+   * the user can choose to dismiss/ignore, or turn it on for the first time.
+   * @param account The particular library for the permission status request.
+   * @param completion Asynchronous handler for the user's decision, or error.
+   */
+  fun requestServerSyncPermissionStatus(account: AccountsControllerType,
+                                        completion: (enableSync: Boolean) -> Unit)
```

Comment:
if two callers both hit this method, will we fire two requests? is that okay for us todo? or should we store completion blocks and just call them when the outstanding request is completed?

Reply:
Thanks for the discussion. It's a pretty rare possibility, but I've gone ahead and made some changes to queue up any possible caller's completions so that multiple network requests are not made. This will save requests, as an example, if someone begins to spam the switch in Settings.